modules/miner/miner.go

```diff
@@ -108,8 +114,9 @@ func New(cs modules.ConsensusSet, tpool modules.TransactionPool, w modules.Walle
 		earliestTimestamp: earliestTimestamp,
 		address:           addr,
 
-		blockMem:  make(map[types.BlockHeader]types.Block),
-		headerMem: make([]types.BlockHeader, headerForWorkMemory),
+		blockMem:   make(map[types.BlockHeader]*types.Block),
+		randTxnMem: make(map[types.BlockHeader]types.Transaction),
```

Comment:
I put this somewhere else, but you probably don't need the whole transaction, just the arb data will be fine.